src/main/java/net/imglib2/converter/RealTypeConverters.java

```diff
@@ -0,0 +1,249 @@
+			return TYPE_IDENTITY;
+		if ( ( inputType instanceof IntegerType ) && ( outputType instanceof IntegerType ) )
+			return integerConverterFactory( inputType, outputType );
+		return floatingPointConverterFactory( inputType, outputType );
+	}
+
+	private static ConverterFactory floatingPointConverterFactory( RealType< ? > inputType, RealType< ? > outputType )
+	{
+		if ( ( inputType instanceof FloatType ) || ( outputType instanceof FloatType ) )
+			return FLOAT;
+		return DOUBLE;
+	}
+
+	private static ConverterFactory integerConverterFactory( RealType< ? > inputType, RealType< ? > outputType )
+	{
+		if ( inputType instanceof LongType || outputType instanceof LongType )
+			return LONG;
+		if ( inputType instanceof IntType || outputType instanceof IntType )
+			return INTEGER;
+		if ( inputType instanceof GenericShortType && outputType instanceof GenericShortType )
+			return SHORT;
+		if ( inputType instanceof GenericByteType && outputType instanceof GenericByteType )
+			return BYTE;
+		if ( inputType instanceof BooleanType && outputType instanceof BooleanType )
+			return BOOLEAN;
+		if ( inputType instanceof GenericIntType && smallerThanInt( outputType ) ||
+				smallerThanInt( inputType ) && outputType instanceof GenericIntType )
+			return INTEGER;
+		return LONG;
+	}
+
+	static boolean smallerThanInt( RealType< ? > variable )
+	{
+		final boolean lowerBound = variable.getMinValue() >= Integer.MIN_VALUE;
+		final boolean upperBound = variable.getMaxValue() <= Integer.MAX_VALUE;
+		return lowerBound && upperBound;
+	}
+
+	private static ConverterFactory TYPE_IDENTITY = new ConverterFactory( TypeIdentity.class );
+
+	private static ConverterFactory DOUBLE = new ConverterFactory( DoubleConverter.class );
+
+	private static ConverterFactory FLOAT = new ConverterFactory( FloatConverter.class );
+
+	private static ConverterFactory INTEGER = new ConverterFactory( IntegerConverter.class );
+
+	private static ConverterFactory LONG = new ConverterFactory( LongConverter.class );
+
+	private static ConverterFactory BYTE = new ConverterFactory( ByteConverter.class );
+
+	private static ConverterFactory SHORT = new ConverterFactory( ShortConverter.class );
+
+	private static ConverterFactory BOOLEAN = new ConverterFactory( BooleanConverter.class );
+
+	/**
+	 * The class {@link ConverterFactory} encapsulates the {@link ClassCopyProvider}
+	 * that is used to create a {@link Converter}.
+	 */
+	private static class ConverterFactory
+	{
+		private final ClassCopyProvider< Converter > provider;
+
+		ConverterFactory( Class< ? extends Converter > converterClass )
+		{
+			this.provider = new ClassCopyProvider<>( converterClass, Converter.class );
+		}
+
+		public Converter create( RealType< ? > in, RealType< ? > out )
+		{
+			return provider.newInstanceForKey( Arrays.asList( in.getClass(), out.getClass() ) );
+		}
+	}
+
+	public static class DoubleConverter implements Converter< RealType< ? >, RealType< ? > >
```

Comment:
With these classes being `public` could you add javadoc, please.
Can be very minimal, e.g.,
```java
/**
 * Converts via {@link RealType#getRealDouble}.
 */
```
 is better than nothing.

Reply:
done